### ISSUE TITLE
MacOS M3 could not launch process: stub exited while waiting for connection: exit status 0

### ISSUE THREAD

**Issue by exosyphon** [go-delve/delve#3804]
I have run through many different steps to getting dlv installed and configured inside Neovim. At this point I cannot run dlv even on the CLI. It is likely user error but I simply don't know what else to do at this point and can use some help since I am new to Golang.

1. What version of Delve are you using (`dlv version`)? 
Delve Debugger
Version: 1.23.0
Build: $Id: e673f2da02185a1c2b6dea52842b0a8c7e7a03b9 $

2. What version of Go are you using? (`go version`)?
go version go1.23.0 darwin/arm64

3. What operating system and processor architecture are you using?
Apple M3 Pro (Apple Silicon)
Sonoma 14.6.1

4. What did you do?
I used asdf to install golang (confirmed that it is arm64 using go version)
installed dlv using these commands after setting these environment variables in my ~/.zshrc:
export GOOS="darwin"
export GOARCH="arm64"
export GOPATH="$HOME/go"
export PATH="$GOPATH/bin:$PATH"

go install github.com/go-delve/delve/cmd/dlv@latest
go install github.com/aarzilli/gdlv@latest

I also already uninstalled and reinstalled xcode and ran through each of the MacOS commands:
xcode-select --install
sudo /usr/sbin/DevToolsSecurity -enable
sudo dscl . append /Groups/_developer GroupMembership $(whoami)

5. What did you expect to see?
When I run dlv debug hello.go to not see an error

6. What did you see instead?
could not launch process: stub exited while waiting for connection: exit status 0

**Comment by aarzilli** [comment]
There is a plethora of similar issues: https://github.com/go-delve/delve/issues?q=is%3Aissue+is%3Aclosed+debugserver++%22stub+exited%22
It's some kind of configuration problem, possibly caused by some amd64 emulation layer running without your knowledge or possibly caused by a broken installation of xcode. The fact that you are explicitly setting GOOS and GOARCH is suspect, these should never be set manually unless you are cross-compiling.

**Comment by exosyphon** [comment]
I was able to figure this out. 

I ran `arch` and the output was `i386` in tmux and Kitty. This should have been `arm64`. I confirmed by running `arch` in Terminal and got `arm64`.

I believe when I initially installed Kitty initially it picked up the wrong architecture from iTerm2.

I re-installed Kitty from inside Terminal:
`curl -L https://sw.kovidgoyal.net/kitty/installer.sh | sh /dev/stdin`

Then checked `arch` was correct. 

I had to re-create my sessions inside tmux since I use resurrect.

Once I did all that I was able to run `dlv debug hello.go`.

Hope this helps someone in the future.
